### PR TITLE
Fill out dashboards for precise-code-intel services

### PR DESCRIPTION
**Pull request by efritz**
This PR adds metrics emitted by the precise-code-intel services to the monitoring config.

There are some TODOs in the monitoring definitions right now that I could use some guidance on.

Screen Shot 2020-05-08 at 11 50 26 AM
Screen Shot 2020-05-08 at 11 50 22 AM
Screen Shot 2020-05-08 at 11 50 18 AM

### CODE REVIEW

File: monitoring/precise_code_intel_bundle_manager.go [hunk 1]
```diff
@@ -6,6 +6,143 @@ func PreciseCodeIntelBundleManager() *Container {
...
+						{
+							Name:            "disk_space_remaining",
+							Description:     "disk space remaining by instance",
+							Query:           `(src_disk_space_available_bytes / src_disk_space_total_bytes) * 100`,
+							DataMayNotExist: true,
+							Warning:         Alert{LessOrEqual: 25},
+							Critical:        Alert{LessOrEqual: 15},
+							PanelOptions:    PanelOptions().LegendFormat("{{instance}}").Unit(Percentage),
+							PossibleSolutions: `
+								- **Provision more disk space:** Sourcegraph will begin deleting the oldest uploaded bundle files at 10% disk space remaining.
```
Comment (slimsag) [inline monitoring/precise_code_intel_bundle_manager.go:66]: Is this true? Does it also use 10%?

Comment (efritz) [reply monitoring/precise_code_intel_bundle_manager.go:66]: Mmhmm: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/precise-code-intel-bundle-manager/env.go#L16

**Comment by slimsag** [comment]
For all of the Critical alerts you have defined, are you extremely confident that if they fire admins should be e.g. paged via OpsGenie and contact us to resolve the issue ASAP? I suspect for a few you may not be, in which case I would just keep the Warning ones for now.

Critical -> site admins will be paged as with any other service outage in that company

Warning -> May be worth looking into, someone may get an email about it.

File: monitoring/precise_code_intel_worker.go [hunk 2]
```diff
@@ -6,6 +6,67 @@ func PreciseCodeIntelWorker() *Container {
...
+						{
+							Name:              "upload_queue_size",
+							Description:       "upload queue size",
+							Query:             `max(src_upload_queue_uploads_total)`,
+							DataMayNotExist:   true,
+							Warning:           Alert{GreaterOrEqual: 100},
+							PanelOptions:      PanelOptions().LegendFormat("uploads queued for processing"),
+							PossibleSolutions: "none",
+						},
+						{
```
Comment (slimsag) [inline monitoring/precise_code_intel_worker.go:22]: sum(increase(src_upload_queue_uploads_total[5m])) / sum(increase(src_upload_queue_processor_total[5m]))

File: monitoring/precise_code_intel_bundle_manager.go [hunk 3]
```diff
@@ -6,6 +6,137 @@ func PreciseCodeIntelBundleManager() *Container {
...
+							`,
+						},
+					},
+					{
+						{
+							Name:              "cache_cost",
+							Description:       "cache cost",
```
Comment (slimsag) [inline monitoring/precise_code_intel_bundle_manager.go:71]: i have no idea what this is

**Comment by efritz** [comment]
Merging this now so we can monitor some things in dot-com ahead of the branch cut. Will respond to additional comments post-merge (and intend to tackle all of the TODOs before branch cut)

**Pull request merged**